File: Linked List, Hash/leetcode/1189_Maximum_number_of_ballloons/maximum_number_of_balloons.cpp
#include <string>
using namespace std;

class Solution {
public:
    int maxNumberOfBalloons(string text) {
        double cnt[] = {0,0,0,0,0};
        double result = 100000;
        char eng[] = {'b', 'a', 'l', 'o', 'n'}; //ba ll oo n
        double values[] = {1, 1, 0.5, 0.5, 1};
        for (auto val: text) {
            for (int i=0; i<5; i++) {
                if (val == eng[i]) {
                    cnt[i] += values[i];
                    break;
                }
            }
        }
        for (auto n: cnt) {
            if (result > n) result = n;
        }
        return int(result);
    }
};

File: Linked List, Hash/leetcode/1189_Maximum_number_of_ballloons/maximum_number_of_balloons.py
class Solution:
    def maxNumberOfBalloons(self, text: str) -> int:
        cnt = [0, 0, 0, 0, 0]
        result = 0
        eng = ['b', 'a', 'l', 'o', 'n']
        values = [1, 1, 0.5, 0.5, 1]
        for i in range(len(text)) :
            for j in range(5) :
                if text[i] == eng[j] :
                    cnt[j] += values[j]
        
        result = int(min(cnt))

        return result;